docs/function-templates/python.md

```diff
@@ -1,160 +1,379 @@
-any Python project. For now, we will ignore the `func.yaml` file, and just
-say that it is a configuration file that is used when building your project.
-If you're really interested, check out the [reference doc](../reference/func_yaml.md).
-To learn more about the CLI and the details for each supported command, see
-the [CLI Commands document](../reference/func.md).
+### Core Methods
 
-## Running the function locally
+Your function class can implement several optional methods:
 
-To run a function, you'll first need to build it. This step creates an OCI
-container image that can be run locally on your computer, or on a Kubernetes
-cluster.
+#### `handle(self, scope, receive, send)`
+The main request handler following ASGI protocol. This async method processes all HTTP requests except health checks.
 
+```python
+async def handle(self, scope, receive, send):
+    """Handle HTTP requests."""
+    # Process the request
+    await send({
+        'type': 'http.response.start',
+        'status': 200,
+        'headers': [[b'content-type', b'text/plain']],
+    })
+    await send({
+        'type': 'http.response.body',
+        'body': b'Hello, World!',
+    })
 ```
-❯ func build
+
+#### `start(self, cfg)`
+
+Called when a function instance starts (e.g., during scaling or updates). Receives configuration as a dictionary.
+
+```python
+def start(self, cfg):
+    """Initialize function with configuration."""
+    self.debug = cfg.get('DEBUG', 'false').lower() == 'true'
+    logging.info("Function initialized")
 ```
 
-After the function has been built, it can be run locally.
+#### `stop(self)`
 
+Called when a function instance stops. Use for cleanup operations.
+
+```python
+def stop(self):
+    """Clean up resources."""
+    # Close database connections, flush buffers, etc.
+    logging.info("Function shutting down")
 ```
-❯ func run
+
+#### `alive(self)` and `ready(self)`
+
+Health check methods exposed at `/health/liveness` and `/health/readiness`:
+
+```python
+def alive(self):
+    """Liveness check."""
+    return True, "Function is alive"
+
+def ready(self):
+    """Readiness check."""
+    if self.database_connected:
+        return True, "Ready to serve"
+    return False, "Database not connected"
 ```
 
-Functions can be invoked with a simple HTTP request.
-You can test to see if the function is working by using your browser to visit
-http://localhost:8080. You can also access liveness and readiness
-endpoints at http://localhost:8080/health/liveness and
-http://localhost:8080/health/readiness. These two endpoints are used
-by Kubernetes to determine the health of your function. If everything
-is good, both of these will return `OK`.
+## Local Development
 
-## Deploying the function to a cluster
+### Running Your Function
 
-To deploy your function to a Kubernetes cluster, use the `deploy` command.
+```bash
+# Build and run in a container (default)
+func run --builder=host
 
+# Force rebuild even if no changes detected
+func run --build
```

Comment:
I find this confusing:
first it says `Build and run in a container (default)`
but then explicitly does opposite `func run --builder=host`